server/jetstream_api.go

```diff
@@ -2341,6 +2361,92 @@ func (s *Server) jsLeaderServerStreamMoveRequest(sub *subscription, c *client, _
 
 	cfg.Placement = origPlacement
 
+	s.Noticef("Requested move of: R=%d stream %s in account %s from old peer set %+v to new peer set %+v",
+		cfg.Replicas, streamName, accName, s.peerSetToNames(currPeers), s.peerSetToNames(peers))
+
+	// execute on a go routine for route or gateway, as this may block waiting from responses from other servers and
+	// therefore block all route/gateway traffic.
+	if c.kind != ROUTER && c.kind != GATEWAY {
```

Comment:
Always wonder if we need LEAF in those decisions?

Reply:
I copied that from update and just did the search for this check:
update only checks for router/Gateway, direct get/next checks leaf too.
```
    Occurrences of 'c.kind != ROUTER && c.kind != GATEWAY' in Project with mask '*.go'
Found Occurrences in Project with mask '*.go'  (6 usages found)
    nats-server  (6 usages found)
        server  (6 usages found)
            consumer.go  (1 usage found)
                2611 inlineOk := c.kind != ROUTER && c.kind != GATEWAY && c.kind != LEAF
            jetstream_api.go  (4 usages found)
                732 if c.kind != ROUTER && c.kind != GATEWAY {
                1398 if c.kind != ROUTER && c.kind != GATEWAY {
                2369 if c.kind != ROUTER && c.kind != GATEWAY {
                2452 if c.kind != ROUTER && c.kind != GATEWAY {
            stream.go  (1 usage found)
                3299 inlineOk := c.kind != ROUTER && c.kind != GATEWAY && c.kind != LEAF
```